cli/src/auth/jwt.rs

```diff
@@ -0,0 +1,79 @@
+/// JWT token parsing.
+use std::collections::HashSet;
+use std::string::FromUtf8Error;
+
+use base64::engine::general_purpose::STANDARD;
+use base64::prelude::*;
+use base64::DecodeError;
+use serde::Deserialize;
+use serde_json::Error as JsonError;
+use thiserror;
+
+/// Get user roles from a bearer token without performing validation.
+pub fn user_roles(bearer: &str) -> Result<HashSet<RealmRole>, JwtError> {
+    // Extract the base64 payload.
+    let (_, payload_base64, _) = parts(bearer)?;
```

Comment:
If the signature is thrown away here and no validation performed, couldn't someone update the payload to claim any role they wanted? That may not matter for `vulnreach` in particular, but this code could easily be relied on to restrict functionality by role for other/future features.

Reply:
> but this code could easily be relied on to restrict functionality by role for other/future features

CLI will never use roles as a security control because CLI is an open source client. The only purpose for looking at user roles in CLI is optimization (to avoid making requests that we know will fail).

And there is little risk of anyone copying this code to use it in a place where security actually matters. First, because the function comment clearly states "without performing validation". And second, because we already have an internal crate dedicated to this purpose of validating these tokens.